pkg/cluster/reconcile.go

```diff
@@ -151,6 +151,29 @@ func (c *Cluster) removeOneMember() error {
 }
 
 func (c *Cluster) removeDeadMember(toRemove *etcdutil.Member) error {
+	if c.cluster.Spec.SelfHosted != nil {
+		selectedNodes, err := c.selectMasterNodes()
+		if err != nil {
+			return err
+		}
+
+		// If we do not have enough master nodes, we should simply wait for the old node
+		// to be online again.
+		//
+		// Removing the etcd member will not help us to recover the cluster to the desired
+		// number of members, since we do not have enough master nodes for self hosted etcd.
+		//
+		// Instead, there is a large chance that the old node is taken offline for maintenance
+		// or experiencing temporary network partition. When it comes back, it will recover itself
+		// since we persist data for self hosted case.
+
+		if nodeNum := len(selectedNodes); nodeNum < c.cluster.Spec.Size {
```

Comment:
Okay so the master nodes selected by `selectMasterNodes()` are to be considered alive,
 i.e condition `Ready=True` only, and all other conditions false.
And nodes not selected could either be temporarily offline(Condition: `Ready=False/Unkown`), or have some permanent issue(Condition: `OutOfDisk=True`).
Does that sound right?
https://kubernetes.io/docs/concepts/architecture/nodes/#condition